packages/zowe-explorer/src/Profiles.ts

```diff
@@ -1237,6 +1237,120 @@ export class Profiles extends ProfilesCache {
+                    serviceProfile.name
+                )
+            );
+            return;
+        }
+        switch (true) {
+            case ProfilesUtils.isProfileUsingBasicAuth(serviceProfile): {
+                let loginOk = false;
+                if (loginTokenType && loginTokenType.startsWith("apimlAuthenticationToken")) {
+                    loginOk = await ZoweVsCodeExtension.loginWithBaseProfile(serviceProfile, loginTokenType, node, zeInstance, this);
+                } else {
+                    loginOk = await this.loginWithRegularProfile(serviceProfile, node);
+                }
+
+                if (loginOk) {
+                    Gui.showMessage(
+                        localize(
+                            "handleSwitchAuthentication.switchFromBasicToTokenAuth.successful",
+                            "Login using token-based authentication service was successful for profile {0}.",
+                            serviceProfile.name
+                        )
+                    );
+                    await Profiles.getInstance().refresh(zeInstance);
+                    await this.basicAuthClearSecureArray(serviceProfile.name);
+                    const updBaseProfile: zowe.imperative.IProfile = {
+                        user: undefined,
+                        password: undefined,
+                    };
+                    node.setProfileToChoice({
+                        ...node.getProfile(),
+                        profile: { ...node.getProfile().profile, ...updBaseProfile },
+                    });
+                } else {
+                    Gui.errorMessage(
+                        localize(
+                            "handleSwitchAuthentication.switchFromBasicToTokenAuth.error",
+                            "Unable to switch to Token-based authentication for profile {0}.",
+                            serviceProfile.name
+                        )
+                    );
+                    return;
+                }
+                break;
+            }
+            case await ProfilesUtils.isUsingTokenAuth(serviceProfile.name): {
+                const profile: string | zowe.imperative.IProfileLoaded = node?.getProfile();
+                const creds = await Profiles.getInstance().promptCredentials(profile, true);
+
+                if (creds != null) {
+                    const successMsg = localize(
+                        "handleSwitchAuthentication.switchFromTokenToBasicAuth.successful",
+                        "Login using basic authentication was successful for profile {0}.",
+                        typeof profile === "string" ? profile : profile.name
+                    );
+                    ZoweLogger.info(successMsg);
+                    Gui.showMessage(successMsg);
+                    await this.tokenAuthClearSecureArray(serviceProfile.name);
+                    await vscode.commands.executeCommand("zowe.extRefresh");
```

Comment:
Instead of doing an extension refresh, should we fire the `onProfilesUpdateEmitter` so that extenders are notified of the update?
Otherwise, running this command after switching auth. methods will reload ZE and its extenders, and they won't be notified of the change.

```suggestion
                    ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(EventTypes.UPDATE);
```